src/app/layout/qgslayoutdesignerdialog.cpp

```diff
@@ -4599,6 +4599,7 @@ bool QgsLayoutDesignerDialog::getPdfExportSettings( QgsLayoutExporter::PdfExport
     mLayout->setCustomProperty( QStringLiteral( "pdfOgcBestPracticeFormat" ), useOgcBestPracticeFormat ? 1 : 0 );
     mLayout->setCustomProperty( QStringLiteral( "pdfExportThemes" ), exportThemes.join( QLatin1String( "~~~" ) ) );
     mLayout->setCustomProperty( QStringLiteral( "pdfLayerOrder" ), geoPdfLayerOrder.join( QLatin1String( "~~~" ) ) );
+    mLayout->setCustomProperty( QStringLiteral( "pdfGroupOrder" ), dialog.geoPdfGroupOrder().join( QLatin1String( "~~~" ) ) );
```

Comment:
I see the use of "~~~" as a conventional delimiter is not new to this pull request but given it is used in another file, shouldn't there be some static QString / method to return that to avoid the "magic string" effect ?

Reply:
@rouault 

Actually turns out that the custom properties already directly support list values, so we can just write the list directly! I won't migrate any of the existing code to change to this, but I've updated the new code accordingly.